sdk/include/psdk/versionhelpers.h

```diff
@@ -148,9 +149,58 @@ IsActiveSessionCountLimited()
 VERSIONHELPERAPI
 IsReactOS()
 {
-    // FIXME: Find a better method!
-    WCHAR szWinDir[MAX_PATH];
-    GetWindowsDirectoryW(szWinDir, _countof(szWinDir));
-    return (wcsstr(szWinDir, L"ReactOS") != NULL);
+    BOOL bResult = FALSE;
+    BOOL bLibFree = FALSE;
+    HMODULE hMod;
+
+    hMod = GetModuleHandleW(L"shell32");
```

Comment:
It would be more reliable to use kernel32.dll instead (which is guaranteed to be always present, and loaded for win32 applications, contrary to shell32), or better, ntdll.dll ?

Also, have you thought about looking at some registry settings as an alternative/simpler way?
For example, the following registry key/value:
`HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion` value `ProductName`

Reply:
The entries in the registry can be easily changed by the user or AD administrator.
Is this secure enough?

Reply:
If you think it's better to have something that always tells "it's ReactOS" even if the user masquerades it by changing the registry entries manually, then it may be better to stay by checking the vendor of a core system dll indeed.

Reply:
This is completely not about security, the only reason this exists is for unittests.
So a simple registry check on product name is more than enough, and way simpler.

Reply:
It was not obvious that these functions are only for unit tests. Why is this file not in the unit tests directory tree?

Currently, ReactOS is a replica of MS Windows. In the distant future, could ReactOS contain features that Windows does not? Application developers who want to use these features would appreciate such a public feature.

Reply:
> It was not obvious that these functions are only for unit tests. Why is this file not in the unit tests directory tree?
> 
> Currently, ReactOS is a replica of MS Windows. In the distant future, could ReactOS contain features that Windows does not? Application developers who want to use these features would appreciate such a public feature.

We do not support feature detection for stuff like that.
If this ever becomes a thing we might want to support, there should be checks on the new functionality, and not on what OS the code is running. (As should all things like this, but that is sadly not always possible)

Reply:
> It was not obvious that these functions are only for unit tests. Why is this file not in the unit tests directory tree?

I assumed you would have at least checked where a function is used before you start modifying it.
Did you check that this works at all?

Reply:
> I assumed you would have at least checked where a function is used before you start modifying it. Did you check that this works at all?

I have not checked if the IsReactOS function is used anywhere.

Whether this function works? With a miniprogram that prints the return value to the console. However still with the shell32.dll.
Now I have detected the problem with the VersionInformation resource.
shell32.dll => english => 040904B0
ntdll.dll => neutral => 000004B0

Ergo, I still have to change that.

Reply:
> > I assumed you would have at least checked where a function is used before you start modifying it. Did you check that this works at all?
> 
> I have not checked if the IsReactOS function is used anywhere.
> 
> Whether this function works? With a miniprogram that prints the return value to the console. However still with the shell32.dll. Now I have detected the problem with the VersionInformation resource. shell32.dll => english => 040904B0 ntdll.dll => neutral => 000004B0
> 
> Ergo, I still have to change that.

Right, but please re-read my comment:
In my opinion this is the wrong way to go, reading the registry is simpler, and more than sufficient for our needs.